libs/cardano-ledger-core/src/Cardano/Ledger/Coin.hs

```diff
@@ -54,6 +56,7 @@ import Cardano.Ledger.Compactible
 import Control.DeepSeq (NFData)
 import Data.Aeson (FromJSON, ToJSON)
 import Data.Coerce (coerce)
+import Data.Foldable (Foldable (..))
```

Comment:
I suggest doing this instead, because it does not require hiding of `Foldable` from `Prelude`, which seems very strange, unless you know about this new re-export from `foldl'` fro Prelude:
```suggestion
import qualified Data.Foldable as F (foldl')
```

And then using it can be used qualified as `F.foldl'`. This is the easiest way to avoid warnings in newer ghc versions.

Reply:
I made this change but I'm not sure I like it. Using a qualified version of a widely-used function like `foldl'` looks strange to me, because it makes it seem like it's a special version of `foldl'` instead of the regular one.

I added a comment to the import of `Data.Foldable` saying, "Drop this when ghc >= 9.10". If we added a similar comment to the import of `Prelude` it would seem less strange, I think.

However, in the end, either way is fine with me.

Reply:
>  I'm not sure I like it

The approach I've suggested have been used throughout cardano-ledger and cardano-base. Consistency should take precedence over preference.
Also, no comments are needed. Instead I'd suggest adding a task to this ticket: https://github.com/IntersectMBO/cardano-ledger/issues/4798